Meta/build-dictionaries.py

```diff
@@ -0,0 +1,227 @@
+#!/usr/bin/env python3
+
+'''
+
+This writes a dictionary file to Base/res/dictionaries/wordnet.bin
+
+The file contains words from the WordNet database, in a format
+designed to be loaded by the Dictionary app, to provide fast
+searching, and to be small and simple.
+
+Dictionary .bin file format:
+
+[Header] @ 0
+u32 num-words
+u32 max-word-length (= 16)
+u32 index-offset
+u32 word-data-index-offset
+
+[Index] @ index-offset
+  Fixed-length ASCII strings for all words,
+  tightly packed, padded with zeroes.
+
+[Word data index] @ word-data-index-offset
+(num-words times):
+  u32 word-data-offset
+
+[Word data]
+  u8  num-senses
+  (num-senses times):
+    u8  sense-number
+    u32 definition-offset
+
+[Definition data] @ definition-offset
+  u8 part-of-speech
+  asciz definition
+
+'''
+
+import struct
+import sqlite3
+import itertools
+import subprocess
+import collections
+import re
+import os
+
+
+def align(f, n):
+    while f.tell() % n:
+        f.write(b'\0')
+
+
+# Download the WordNet database (100MB)
+url = 'https://pilotfiber.dl.sourceforge.net/project/wnsql/wnsql3/sqlite/3.1/sqlite-31.db.zip'
```

Comment:
I'm not super convinced that this use of the WordLite database complies with their license? For instance, I don't see the License reproduced at all in this binary packer for the data within.

https://wordnet.princeton.edu/license-and-commercial-use

> When writing a paper or producing a software application, tool, or interface based on WordNet, it is necessary to properly cite the source. 

https://wordnet.princeton.edu/citing-wordnet

Reply:
I know, I mentioned it in the PR description. I think the about box would probably be the best place for this.

Reply:
Help menu item opening the license TextEditor might be good for now, though? I've added a checkbox to the description so we don't forget.

Reply:
I would say the about box is a good a place as any, yeah.

Reply:
OK, license added. I put it as its own Help menu entry because the About box isn't currently customizable and I didn't fancy trying to make that change as well 🙂